src/lib.rs

```diff
@@ -41,13 +42,15 @@ extern crate tempfile;
 extern crate zdaemon;
 extern crate zfilexfer;
 extern crate hostname;
+extern crate interfaces;
+extern crate ipnetwork;
 
 #[macro_use]
 mod ffi_helpers;
 mod command;
 mod project;
 mod directory;
-mod error;
+pub mod error;
```

Comment:
I hadn't noticed that we are reexporting `Error`. I had to make it public because trying to build the `hello` example was failing due to `error` being private.

```
error: module `error` is private
 --> src/main.rs:9:22
  |
9 |     let interfaces = needobj!(data => "/_telemetry").unwrap();
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate
```

This is happening because we are using `$crate::error::Error::Generic` in the macros instead of `$crate::Error::Generic`.